spec/ics-2-consensus-verification/README.md

```diff
@@ -0,0 +1,190 @@
+---
+
+ics: 2
+title: Consensus Verification
+stage: proposal
+category: ibc-core
+requires: 23
+required-by: 3
+author: Juwoon Yun <joon@tendermint.com>, Christopher Goes <cwgoes@tendermint.com>
+created: 2019-02-25
+modified: 2019-04-02
+
+---
+
+## Synopsis
+
+This standard specifies the properties that consensus algorithms of chains implementing IBC are 
+expected to satisfy. The properties are needed for efficient and safe verification in the higher
+level protocol abstractions. The algorithm who uses these properties to verify substates of 
+another chain is referred to as a "light client".
+
+## Specification
+
+### Motivation
+
+`Header` defines required properties of the blockchain on the network. The implementors can 
+check whether the consensus that they are using is qualified to be connected to the network or 
+not. If not, they can modify the algorithm or wrap it with additional logic to make it 
+compatible with the specification. It also provides base layer for the protocol that the other 
+components can rely on.
+
+### Desired Properties
+
+* Blockchains, defined as an infinite list of `Header` starting from its genesis, is linear; no 
+conflicting `Header`s can be both validated, thus no data can be rewritten after it has been 
+committed. Two `Header`s are conflicting when both has same height but not equal.
+
+* Verifiers can verify future `Header`s using an existing `RootOfTrust`. When the verifier 
+validates it, the verified header is in the canonical blockchain.
+
+* `RootOfTrust`s contains an accumulator root(ICS23) that the other logics can verify whether 
+key-value pairs exists or not with it.
+
+### Technical Specification
+
+#### Definitions
+
+* `RootOfTrust` is a blockchain commit which contains an accumulator root and the requisite 
+  state to verify future roots, stored in one blockchain to verify the state of the other.
+  Defined as 2-tuple `(v :: Header -> (Error|RootOfTrust), r :: AccumulatorRoot)`, where
+    * `v` is the verifier, proves child `Header.p` and returns the updated `RootOfTrust`
+    * `r` is the `AccumuatorRoot`, used to prove internal state
+
+* `Header` is a blockchain header which provides information to update `RootOfTrust`, 
+  submitted to one blockchain to update the stored `RootOfTrust`.
+  Defined as 3-tuple `(p :: HeaderProof, v :: Maybe<Header -> (Error|RootOfTrust)>,
+  r :: AccumulatorRoot)`, where
+    * `p` is the commit proof used by `RootOfTrust.v` to verify
+    * `v` is the new verifier, if needed to be updated
+    * `r` is the new `AccumulatorRoot` which will replace the existing one
+ 
+* `Consensus` is a blockchain consensus algorithm which generates valid `Header`s.
+  Defined as a function `RootOfTrust -> [Transaction] -> Header`
+
+* `Blockchain` is a subset of `(Consensus, RootOfTrust, [Header])`, generated 
```

Comment:
Tuples can't have subsets, since they aren't sets - what is the set being referred to here?